### DISCUSSION TITLE
Update session config at runtime

### DISCUSSION THREAD

**Discussion by RGregat** [sceneview/sceneform-android#81] [Q&A]
I'm not 100% sure if it is a bug or a feature. 
I can change the Plane-Detection during the lifetime of a session. If I need it, I enable it, if not, disabled. I want to do the same thing with the DepthMode. But that is not working. Where is the difference between Plane-Detection and DepthMode? Why is it possible to change one part but not the other part during the lifetime of a session.

I asked the same question on Stackoverflow: https://stackoverflow.com/questions/67921780/update-session-config-at-runtime

**Comment by ThomasGorisse** [comment]
I don't have the answer on this.
You should definitely ask this question on the [ARCore repo](https://github.com/google-ar/arcore-android-sdk).
They are quite active for answering.

**Reply by RGregat** [reply]
Got it, thought the same :)

**Reply by RGregat** [reply]
Ok I asked the question on the ARCore Repo: https://github.com/google-ar/arcore-android-sdk/issues/1243

**Reply by ThomasGorisse** [reply]
BTW, how do you change the config during the session lifetime?

You just call `session.configure(config);` or you have a special trick since I thought the second call to `configure()` was not working?

It would be enormously helpful for Augmented Images since we have to call `config.setAugmentedImageDatabase()` after a long Bitmap loading at the session init time.

**Reply by RGregat** [reply]
Well, that is what I'm doing. I just thought, give it a shot and see what happens :P. 

**Activate**
```java
Optional.ofNullable(mArFragment.getArSceneView().getSession())
   .ifPresent(session -> {
      Config config = session.getConfig();
      config.setPlaneFindingMode(Config.PlaneFindingMode.HORIZONTAL_AND_VERTICAL);
      config.setDepthMode(Config.DepthMode.DISABLED);
      mArFragment.getArSceneView().getSession().configure(config);
});
```

**Deactivate**
```java
Optional.ofNullable(mArFragment.getArSceneView().getSession())
    .ifPresent(session -> {
       Config config = session.getConfig();
       config.setPlaneFindingMode(Config.PlaneFindingMode.DISABLED);
       if(session.isDepthModeSupported(Config.DepthMode.AUTOMATIC))
          config.setDepthMode(Config.DepthMode.AUTOMATIC);
       mArFragment.getArSceneView().getSession().configure(config);
});
```

I also had in mind that this shouldn't work. And maybe that is the case? To verify the toggle, I started the new performance overlay and checked the Plane Detection CPU usage. And to my surprise the CPU usage is going down to 0.0% if I disable the Plane-Detection and goes back to 1-5% (depending on the Smartphone used) if I enable it again.
